plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt

```diff
@@ -77,6 +79,14 @@ open class KtlintCheckTask @Inject constructor(
         }
     }
 
+    private val newFileProperty: () -> RegularFileProperty =
+        if (GradleVersion.current() >= GradleVersion.version("5.0")) {
+            val method = JavaReflectionUtil.method(objectFactory, RegularFileProperty::class.java, "fileProperty");
+            { method.invoke(objectFactory) }
+        } else {
+            ::newOutputFile
+        }
```

Comment:
Um, I think there's a better way to do this.

Reply:
@big-guy We can't due to the issue outlined here:
https://github.com/JLLeitschuh/ktlint-gradle/issues/185

Reply:
@JLLeitschuh We might just upgrade the plugin project, but leaving the root project with all the samples as is. Seems to work for me locally. Or am I missing something here?